api_views/books.py

```diff
@@ -77,4 +84,28 @@ def get_by_title(book_title):
                 }
                 return Response(json.dumps(responseObject), 200, mimetype="application/json")
             else:
-                return Response(error_message_helper("Book not found!"), 404, mimetype="application/json")
\ No newline at end of file
+                return Response(error_message_helper("Book not found!"), 404, mimetype="application/json")
+
+
+def delete_book(book_id):
+    req_data = request.json()
+    try:
+        jsonschema.validate(req_data, delete_book_schema)
+    except:
+        return Response(error_message_helper("Invalid JSON format."), 500, mimetype="application/json")
+
+    resp = token_validator(request.headers.get('Auth-Token'))
+    if "Invalid" in resp:
+        return Response(error_message_helper("Authorization failed"), 404, mimetype="application/json")
+
+    usr = User.query.filter(User.username == resp).first()
+    if not usr:
+        return Response(error_message_helper("User not found."), 400, mimetype="application/json")
+
+    b = Book.query.get(book_idd)
+    if b.user_id != usr.id:
+        return Response(error_message_helper("Unauthorized access."), 403, mimetype="application/json")
+
+    db.delete(b)
```

Comment:
Incorrect usage of db.delete() - should be db.session.delete(b) followed by db.session.commit()